File: Data Structures & Algorithms/remove-element/submission-2.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            if nums[l] == val:
                while r >= l and nums[r] == val:
                    r -= 1
                nums[l] = nums[r]
                r -= 1
            if r >= l:
                l += 1
        return l

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> m; // num -> cnt
        for (auto num : nums) {
            if (m.contains(num)) {
                m[num]++;
            } else {
                m[num] = 1;
            }
        }

        priority_queue<pair<int, int>> pq;
        for (auto [num, cnt] : m) {
            pq.push({cnt, num});
        }

        vector<int> ret;
        while(k--) {
            auto [cnt, num] = pq.top();
            pq.pop();
            ret.push_back(num);
        }
        return ret;
    }
};
